src/main/java/algorithms/curated170/medium/ShortestWordDistance2.java

```diff
@@ -0,0 +1,55 @@
+package algorithms.curated170.medium;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class ShortestWordDistance2 {
+
+    public class WordDistance {
+
+        HashMap<String, List<Integer>> wordIndices;
+
+        public WordDistance(String[] words) {
+            wordIndices = new HashMap<String, List<Integer>>();
+
+            for (int i = 0; i < words.length; i++) {
+                if (!wordIndices.containsKey(words[i])) {
+                    wordIndices.put(words[i], new ArrayList<Integer>());
+                }
+                wordIndices.get(words[i]).add(i);
+            }
+        }
+
+        public int shortest(String word1, String word2) {
+            List<Integer> ind1 = wordIndices.get(word1);
+            List<Integer> ind2 = wordIndices.get(word2);
+            int minDist = wordIndices.size();
```

Comment:
We should change this. Because "two words" will constantly change at every **shortest(String word1, String word2)** call. It is why we have to reset the **minDist** too. We can do it using **words.length**, but this is easier:
            
```suggestion
           int minDist = Integer.MAX_VALUE;
```